When cross compiling we need to change the way the python support is
passed into configure.  This allows us to pass the base directory, but the
system will still attempt to find the executable to run.

Signed-off-by: Mark Hatle <mark.hatle@windriver.com>

Index: libxml2-2.7.8/configure.in
===================================================================
--- libxml2-2.7.8.orig/configure.in
+++ libxml2-2.7.8/configure.in
@@ -702,17 +702,17 @@ PYTHON_SITE_PACKAGES=
 PYTHON_TESTS=
 pythondir=
 if test "$with_python" != "no" ; then
-    if test -x "$with_python/bin/python"
+    if test -d "$with_python" -a -x "$with_python/bin/python"
     then
         echo Found python in $with_python/bin/python
         PYTHON="$with_python/bin/python"
     else
-        if test -x "$with_python/python.exe"
+        if test -d "$with_python" -a -x "$with_python/python.exe"
         then
             echo Found python in $with_python/python.exe
             PYTHON="$with_python/python.exe"
         else
-            if test -x "$with_python"
+            if test ! -d "$with_python" -a -x "$with_python"
             then
                 echo Found python in $with_python
                 PYTHON="$with_python"
@@ -734,8 +734,7 @@ if test "$with_python" != "no" ; then
     fi
     if test "$PYTHON_VERSION" != ""
     then
-	if test -r $with_python/include/python$PYTHON_VERSION/Python.h -a \
-	   -d $with_python/lib/python$PYTHON_VERSION/site-packages
+	if test -r $with_python/include/python$PYTHON_VERSION/Python.h
 	then
 	    PYTHON_INCLUDES=$with_python/include/python$PYTHON_VERSION
 	    PYTHON_SITE_PACKAGES=$libdir/python$PYTHON_VERSION/site-packages
